### ISSUE TITLE
Official Documentation Example for Function Calling Does Not Work Post Major Update

### ISSUE THREAD

**Issue by rafgonlop1** [openai/openai-python#691]
## Issue Description
Following the recent major update to the OpenAI library, the function calling example provided in the official documentation no longer works.

### Steps to Reproduce
1. Visit the official OpenAI documentation on function calling with Chat models [here](https://platform.openai.com/docs/guides/function-calling).
2. Attempt to run the provided example.

### Expected Behavior
The function calling example should execute without errors, demonstrating how to utilize the feature with the latest library version.

### Actual Behavior
When attempting to run the function calling example from the official documentation, the following error occurs: [insert error message]

**Comment by RobertCraigie** [comment]
What error are you seeing?

Running the code sample mentioned in the 2nd step results in this output for me (after modifying it to output json)
```json
{
  "id": "chatcmpl-8I2TF5YC6GBhyMXRc6d5bXC5diNKq",
  "choices": [
    {
      "finish_reason": "stop",
      "index": 0,
      "message": {
        "content": "The current weather in San Francisco is 72°F with clear skies. In Tokyo, it is 10°C and cloudy. In Paris, the temperature is 22°C with mostly sunny skies.",
        "role": "assistant",
        "function_call": null,
        "tool_calls": null
      }
    }
  ],
  "created": 1699310093,
  "model": "gpt-3.5-turbo-1106",
  "object": "chat.completion",
  "system_fingerprint": "fp_eeff13170a",
  "usage": {
    "completion_tokens": 39,
    "prompt_tokens": 169,
    "total_tokens": 208
  }
}
```

**Comment by rafgonlop1** [comment]
Hello @RobertCraigie,

You are correct. I had to verify it because my IDE was indicating that there were attributes that did not exist in the classes. I updated my OpenAI library version about an hour ago, and it installed version 1.0.1. I did not bother to check if that could be the problem, but upon rechecking, it led me to version 1.1.1, where they do exist.

Thank you, and apologies for any inconvenience.
